lib/bolt/transport/base.rb

```diff
@@ -0,0 +1,158 @@
+require 'logging'
+
+module Bolt
+  module Transport
+    # This class provides the default behavior for Transports. A Transport is
+    # responsible for uploading files and running commands, scripts, and tasks
+    # on Targets.
+    #
+    # Bolt executes work on the Transport in "batches". To do that, it calls
+    # the batches() method, which is responsible for dividing the list of
+    # Targets into batches according to how it wants to handle them. It will
+    # then call Transport#batch_task, or the corresponding method for another
+    # operation, passing a list of Targets. The Transport returns a list of
+    # Bolt::Result objects, one per Target. Each batch is executed on a
+    # dedicated thread, so many batches may be running in parallel.
```

Comment:
I would mention that the size of this thread pool is controlled by the concurrency option.

Reply:
Updated to clarify that